src/client/pydaos/SConscript

```diff
@@ -0,0 +1,66 @@
+"""Build pydaos client"""
+import daos_build
+import distutils.sysconfig
+
+def build_shim_module(py_env, svars, lib_prefix, version):
+    """Build PyDAOS shim module for the specified python version"""
+
+    new_env = py_env.Clone()
+    extra_flags = ""
+
+    if version == "2.7":
+        tgt_name = 'pydaos_shim_27'
+        base_name = 'pydaos_shim_27'
+    elif version == "3.6":
+        tgt_name = 'pydaos_shim_36'
+        base_name = 'pydaos_shim_36'
+        extra_flags = ' -D__USE_PYTHON3__==1'
+    else:
+        print("Unsupported python version %s" % version)
+        return
+
+    obj = new_env.SharedObject(tgt_name, 'pydaos_shim.c',
+                               LIBS=['daos'],
+                               CC=svars["CC"],
+                               SHLINK=svars["LDSHARED"],
+                               SHLINKFLAGS=[],
+                               SHLIBPREFIX="",
+                               SHLIBSUFFIX=svars["SO"],
+                               CPPPATH=["$CPPPATH",
+                                        distutils.sysconfig.get_python_inc()],
+                               CPPFLAGS=svars["BASECFLAGS"] + " " + \
+                               svars["OPT"] + " " + extra_flags)
+
+    base = new_env.SharedLibrary(target=base_name, source=[obj],
+                                LIBS=['daos'],
+                                CC=svars["CC"],
+                                SHLINK=svars["LDSHARED"],
+                                SHLINKFLAGS=[],
+                                SHLIBPREFIX="",
+                                SHLIBSUFFIX=svars["SO"],
+                                CPPPATH=["$CPPPATH",
+                                         distutils.sysconfig.get_python_inc()],
+                                CPPFLAGS=svars["BASECFLAGS"] + " " + \
+                                svars["OPT"] + " " + extra_flags)
+    install_path = lib_prefix + "/python" + version + "/site-packages"
+    new_env.Install(install_path, base)
+    # For new wrappers, once available
+    # new_env.Install(install_path, "pydaos.py")
+
+def scons():
+    """Execute build"""
+    Import('env', 'prereqs')
+
+    svars = {}
+    for item in ['CC', 'OPT','BASECFLAGS', 'LDSHARED', 'SO']:
+        svars[item] = distutils.sysconfig.get_config_var(item)
+	if svars[item] is None:
+            svars[item] = ""
+
+    py_env = env.Clone()
+
+    build_shim_module(py_env, svars, '$PREFIX/lib/', "2.7")
+    build_shim_module(py_env, svars, '$PREFIX/lib/', "3.6")
```

Comment:
The build error comes from those two lines that try to build for both python 2.7 and 3.6 which are not available on the build node. Any suggestions on how to address that are welcomed.

Reply:
> The build error comes from those two lines that try to build for both python 2.7 and 3.6 which are not available on the build node. Any suggestions on how to address that are welcomed.

You need to add the python-2 and python-3 _devel_ packages to the various build environments.

For the non-RPM builds this means adding the respective packages to the platforms' Dockerfiles in `utils/docker/`.

For the RPM builds, you need to add a `BuildRequires` for the platform's python-2 and python-3 _devel_ packages.  For EL7 for example that is `BuildRequires: python-devel python36-devel`.  I'm not sure what it is for the other platforms.  If they are different from EL7 (most likley) they will need to be guarded by platform-detection macros:

https://github.com/daos-stack/daos/blob/df2e3b4c20db2515ac85de78d298d0d1b8b5cf9e/utils/rpms/daos.spec#L37-L48

Additionally, you will need to add the packages that you added above for all but EL7 to their respective `utils/docker/Dockerfile-rpmbuild` files.

The reason you don't have to do this for EL7's `utils/docker/Dockerfile-mockbuild` is because we have the nice `mock` tool on EL7 that installs all of the dependencies specified in the `daos.spec` file into a chroot before building the RPM in it.  `mock` doesn't exist on SUSE distros unfortunately.  They do have a different tool which supposedly does similar to what mock does but I have not investigated it yet.

Reply:
Is there any way to do it for the default version 3 rather than specifying 3.6 specifically?   Then in SConscript, we can just test what version of python3 we have and build accordingly?  Or is it simpler to require a specific version?

Reply:
Added BuildRequires as Brain suggested.
For use python version major only,  it's feasible to get the running version online and use the major.minor version to name and place the generated shim file properly. Whether we need to enforce certain python version to match certain DAOS feature requirement, need more discussion.

Reply:
> Is there any way to do it for the default version 3 rather than specifying 3.6 specifically?

Not for the EL7 RPM, I'm afraid.  The python36 RPM doesn't have any _generic_ python-3 kind of "provides".

Reply:
Couldn't we use the existence of the `python3` command to determine if python 3.x is installed and then use it to get the minor version:
```
python3 -c 'import sys; print("{}.{}".format(sys.version_info.major, sys.version_info.minor))'
3.4
```